### PR TITLE
Add dict_converter parameter to toJs

### LINKED ISSUE #1529: Passing python dict into a js function
Hi, I just noticed that we now (v0.17.0) convert python dict to a `Map` in js, and it break my previous code.

I have 3 questions:
1. What would be the easiest way to send a dictionary in Python to Js so it will become a regular object. 
2. What is the reason to choose converting `dict` to `Map`? I find it's bit inconvenient if we get a Map instead of an object by default. I mean, `Object`/`{}` is obviously more commonly used than `Map` in JS, wouldn't it be a better default type for converting `dict` in Python?
3. Previously, if we import js function (e.g. `postMessage`) into Python, then send a python dictionary( `js.postMessage({'data': 123})`, it automatically convert the arguments into js object, but now it complains that `JsException: DataCloneError: Failed to execute 'postMessage' on 'DedicatedWorkerGlobalScope': [object Object] could not be cloned.`.

### PR DESCRIPTION
**Pull request by hoodmane**
This resolves #1529 by allowing `dict_converter=Object.fromEntries`.

Perhaps we should rename the options for `toJs` to follow the Javascript naming conventions (so then `to_js` would have `dict_converter` and `toJs` would have `dictConverter`.

### CODE REVIEW

File: src/py/_pyodide/_core_docs.py [hunk 1]
```diff
@@ -144,22 +145,30 @@ def to_js(
...
-        pyproxies: JsProxy, default = None
-            Should be a Javascript ``Array``. If provided, any ``PyProxies`` generated
-            will be stored here. You can later use :any:`destroy_proxies` if you want
-            to destroy the proxies from Python (or from Javascript you can just iterate
-            over the ``Array`` and destroy the proxies).
-
-        create_pyproxies: bool, default=True
-            If you set this to False, :any:`to_js` will raise an error
-
+        obj : Any The Python object to convert
```
Comment (rth) [inline src/py/_pyodide/_core_docs.py:148]: Even, if this seems to currently render fine, there is really a spec for Python docstrings so it should be,
```
x : type
    Description of parameter `x`.
```
with optionally a default value, as it was before. Indentation value for the Description doesn't matter. https://numpydoc.readthedocs.io/en/latest/format.html#parameters

File: src/core/python2js.c [hunk 2]
```diff
@@ -486,6 +511,85 @@ python2js_with_depth(PyObject* x, int depth, JsRef proxies)
...
+}
+
+JsRef
+python2js_custom_dict_converter(PyObject* x,
+                                int depth,
+                                JsRef proxies,
+                                JsRef dict_converter)
+{
+  if (dict_converter == NULL) {
+    return python2js_with_depth(x, depth, proxies);
```
Comment (rth) [inline src/core/python2js.c:572]: Maybe add a comment that by default it converts to a Map?

**Review by rth — COMMENTED** [review]
Thanks! Generally LGTM, except that the formatting of Python docstrings in to_js would need to be reverted. (could `[skip ci]` in commits) (cf comment below)

File: src/py/_pyodide/_core_docs.py (hunk 1 above)
Comment (hoodmane) [reply src/py/_pyodide/_core_docs.py:148]: Oops, my text editor textwrap tool did this.

Comment (hoodmane) [reply src/py/_pyodide/_core_docs.py:148]: This is why I had the blank lines in there before.

**Pull request merged**